src/Microsoft.Framework.Runtime.Loader/LoadContext.cs

```diff
@@ -60,7 +60,7 @@ public static void InitializeDefaultContext(LoadContext loadContext)
         private string GetNativeImagePath(string ilPath)
         {
             var directory = Path.GetDirectoryName(ilPath);
-            var arch = Environment.GetEnvironmentVariable("PROCESSOR_ARCHITECTURE");
+            var arch = IntPtr.Size == 4 ? "x86" : "AMD64";
```

Comment:
I asked @davidfowl about exactly this and he said they used this pattern elsewhere in the stack as well, so there's already debt.  If we actually end up needing to detect this, I would much prefer adding an API to the BCL and reading that vs environment variables.